tests/Browser/Pages/Backend/Categories/AdminListCategoriesTest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Tests\Browser\Pages\Backend\Categories;
+
+use App\Model\Book;
+use App\Model\Category;
+use App\Model\User;
+use App\Model\Donator;
+use Tests\DuskTestCase;
+use Laravel\Dusk\Browser;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Support\Facades\DB;
+use Faker\Factory as Faker;
+
+class AdminListCategoriesTest extends DuskTestCase
+{
+    use DatabaseMigrations;
+
+    /**
+     * make a User with role 1.
+     *
+     * @return void
+     */
+    public function userLogin()
+    {
+        return factory(User::class)->create([
+            'role' => User::ROLE_ADMIN
+        ]);
+    }
+
+    /**
+     * A Dusk test route to page list categories.
+     *
+     * @return void
+     */
+    public function testRouteShowListCategories()
+    {
+        $user = $this->userLogin();
+        $this->browse(function (Browser $browser) use ($user) {
+            $browser->loginAs($user)
+                    ->visit('/admin')
+                    ->clickLink('CATEGORIES')
+                    ->assertPathIs('/admin/categories')
+                    ->assertSee('List Categories');
+        });
+    }
+
+    /**
+     * Test layout of List Categories.
+     *
+     * @return void
+     */
+    public function testLayoutListCategories()
+    {
+        $user = $this->userLogin();
+        $this->browse(function (Browser $browser) use ($user) {
+            $browser->loginAs($user)
+                ->visit('/admin/categories')
+                ->assertSee('List Categories')
+                ->assertSeeLink('Admin')
+                ->assertSee('ID')
+                ->assertSee('Name')
+                ->assertSee('Number of Books');
+        });
+    }
+
+    /**
+     * Create virtual data to test
+     *
+     * @return void
+     */
+    public function makeDataOfListCategories($rows)
+    {
+        $faker = Faker::create();
+
+        factory(Category::class, $rows)->create();
+        $categoryIds = DB::table('categories')->pluck('id')->toArray();
+
+        factory(Donator::class)->create();
+    
+        for ($i = 0; $i <= $rows; $i++)
+        {
```

Comment:
<?php
for ($i = 0; $i < 10; $i++) {
 // for body
}
em fix convention code đi Nghĩa